### DISCUSSION TITLE
From PCB BOM to purchase order [easier?]

### DISCUSSION THREAD

**Discussion by brechtdevuyst** [inventree/InvenTree#2668] [Q&A] [answered]
I have a rather easy task, but find it hard to do in Inventree, maybe someone could help me?

We have a product consisting of x pcb's, for which I uploaded and validated the BOM's already. Now, we want to know which electronical components we need to order to make y of these products. Some stock is already avaible.

I tried solving this problems in 2 ways:

The first one was making a build order in inventree of y products, then since it are all eletronical components, I changed all the stock for it from tracked (somehow this was the default) to untracked. 
Now I can auto allocate stock, but when there isn't enough stock, I have to individually click the blue shopping cart and manually fill in the required quantity before I can add it to a purchase order. We're talking about hundreds of rows here... so not very convenient.

The other way I tried solving it, was by just exporting the BOM, including stock data to excel. The problem here is that only the available stock can be seen, not the amount that is already required for other build orders. It also gets very tidious when combing more than on board, with similar components, because you then have to merge them and remove duplicates. 
I find myself needing to manually keep this master excel file as a stock overview, which undermines the use of inventree for this.

I would love to see 2 improvements: 
For the first problem, I would like to see a shopping cart button that automatically adds all the needed components to a purchase order of that build, taking into account the current (reserved!) stock level.
Another solution for the excel way, could be that I could also export the "required for builds" stock information of that part.

Please correct me if I'm doing it the wrong way/suggest other methods that would make my live more easy ;)

Thanks in advance for the responses.

**Comment by SchrodingersGat** [comment, marked as answer]
@brechtdevuyst good question. The "order parts" workflow is already tagged as [needing work](https://github.com/inventree/InvenTree/issues/1996). One of the specific problems I want to address is the workflow as you have outlined here. Namely, there are a bunch of parts required to be ordered for a build order.

Ideally, a single button on the build order "Order Required Parts" to create and manage required purchase orders.

However the implementation of this has proved difficult and did not make the cut for the latest release. I've got a lot of other stuff on for the next little while, so if someone else is available to put the time into this, it would be great. It would require a pretty in-depth understanding of the API and back-end, but I am happy to provide guidance if needed. Otherwise, it'll be in the "next stable release", in a few months time.

> I changed all the stock for it from tracked (somehow this was the default) to untracked.

You can change the default behaviour in the settings, here:

![image](https://user-images.githubusercontent.com/10080325/155418351-d7f9435e-07ed-414e-a0b7-12df0e4bd5b0.png)
